operators/utils/functions.py

```diff
@@ -290,14 +290,24 @@ def trim_strips(context, frame_start, frame_end, to_trim, to_delete=[]):
         elif s.frame_final_end > trim_start and s.frame_final_start < trim_start:
             s.frame_final_end = trim_start
 
-    for s in to_delete:
-        bpy.context.sequences.remove(s)
-
+    delete_strips(to_delete)
     for s in initial_selection:
         s.select = True
     return {"FINISHED"}
 
 
+def delete_strips(to_delete):
+    """
+    Deletes the list of sequences `to_delete`
+    """
+    if not to_delete:
+        return
+    bpy.ops.sequencer.select_all(action="DESELECT")
+    for s in to_delete:
+        s.select = True
+    bpy.ops.sequencer.delete()
+
+
```

Comment:
This causes errors in latest Blender versions. The VSE's maintainer himself reported the issue. We have to delete sequences from the array, we musn't use the delete operator.

The current function works as expected for me.

Reply:
> The current function/code works as expected for me. Version 2.92.0

I just redownloaded blender both internal and portable to make sure: The current version 2.92.0 does not include all changes made in the power sequencer. You can check it yourself going to the file reproducing the main bug fixed in the last refactor we made.

The changes are in 2.93 alpha (where I noticed the bug).

> The VSE's mantainer himself reported the issue

May you text me a link to see where he says so?
The same method to delete strips is being used currently in 2.93 alpha in the ripple move btw:
https://github.com/GDQuest/blender-power-sequencer/blob/8fc7e4372445e48f7d32fb366804921e9da4ba21/operators/utils/functions.py#L367-L371

Reply:
There you go https://github.com/GDQuest/blender-power-sequencer/issues/447

Reply:
> The current version 2.92.0 (windows) does not include all changes made in the power sequencer.

That's normal, changes happen only when I update the blender code, and for the next release. For example, if I include changes now, they'll be in 2.93 when it comes out.

Reply:
Hmm, it works fine for me with the lastest version of power sequencer (bug fix included)

Reply:
Could you test this in the lastest 2.93 to make sure it's not only me?

Reply:
I hadn't noticed you said it was in 2.93 alpha. If it's only in that version, then please don't do bug fixes for alpha builds of Blender. It's too early to update the add-on. Wait at least for beta.

In 2.93 using the sequencer freezes my computer, so it's not even in a usable state for me.

Reply:
> Hmm, it works fine for me with the lastest version of power sequencer (bug fix included)

Did you read Richard's comment? He told us not to use the delete operator due to an inherent limitation in the Python API. So, even if it works in some contexts, I'd rather follow his lead and avoid keeping code that might cause crashes in some situations, anytime.

Reply:
Yeah so that's why I was asking about what the bug was exactly, and I didn't get the PR.

The bug's not happening on my side. I'm getting your third result with the current master branch.

Here's a screencap of your setup I just reproduced.

![Screenshot from 2021-03-31 17-03-11](https://user-images.githubusercontent.com/12694995/113221814-ff151280-9242-11eb-8e1f-8ba3fac6ffbe.png)

Reply:
I'm on windows 10. And yes, I'm using the current code. Could you try to reproduce this? (To make sure we're using the same).
![column-bug](https://user-images.githubusercontent.com/74924552/113250431-c582ec80-92c0-11eb-8265-475cc07dfd8f.gif)

Reply:
Ah this, I do get the same as you, where it cuts the duration of both strips rather than just the top one!

Can we merge the code except for the delete part for now? And I'll ask Richard about the issue you face with deletion, see if it's a bug in 2.93 alpha, or if not what we can do.

I can cherry-pick changes if that's fine for you, or you can split your PR into two if that's something you're comfortable doing.